Lib/test/test_json/test_tool.py

```diff
@@ -100,9 +100,22 @@ def test_help_flag(self):
         self.assertEqual(err, b'')
 
     def test_sort_keys_flag(self):
-        infile = self._create_infile()
+        infile = self._create_infile(self.data)
         rc, out, err = assert_python_ok('-m', 'json.tool', '--sort-keys', infile)
         self.assertEqual(rc, 0)
         self.assertEqual(out.splitlines(),
                          self.expect_without_sort_keys.encode().splitlines())
         self.assertEqual(err, b'')
+
+    def test_no_ascii_flag(self):
+        data = '{"json": "🐍 and δ"}'
+        expect = textwrap.dedent('''\
+        {
+            "json": "🐍 and δ"
+        }
+        ''')
+        infile = self._create_infile(data)
+        rc, out, err = assert_python_ok('-m', 'json.tool', '--no-ensure-ascii', infile)
```

Comment:
I afraid this fails on non-UTF-8 locale.

Reply:
Does this still fail with Python 3.7? Because in https://docs.python.org/3.7/library/json.html there isn't written anymore that `--ensure-ascii` can cause problems, while you can still find it in e.g. https://docs.python.org/2.7/library/json.html

Reply:
Haven't tested but I assume so. The issue that Python uses the system encoding which cannot represent certain unicode characters. I could imagine a few possible behaviors:

1. json.tool fails when trying to output a character that cannot be represented by the output encoding
2. json.tool always outputs utf-8
3. json.tool attempts to output using the system's encoding. In the case of failure, the exception is handled, the procedure is repeated with `--no-ensure-ascii` disabled.